CMakeLists.txt

```diff
@@ -1,111 +1,104 @@
 add_library(example::example ALIAS example_example)
 
 generate_export_header(
-    example_example
-    BASE_NAME example
-    EXPORT_FILE_NAME "include/example/example_export.h"
+  example_example BASE_NAME example EXPORT_FILE_NAME
+  "include/example/example_export.h"
 )
 
 set_target_properties(
-    example_example PROPERTIES
-    CXX_VISIBILITY_PRESET hidden
-    VISIBILITY_INLINES_HIDDEN YES
-    VERSION "${PROJECT_VERSION}"
-    SOVERSION "${PROJECT_VERSION_MAJOR}"
-    EXPORT_NAME example
-    OUTPUT_NAME example
+  example_example
+  PROPERTIES CXX_VISIBILITY_PRESET hidden
+             VISIBILITY_INLINES_HIDDEN YES
+             VERSION "${PROJECT_VERSION}"
+             SOVERSION "${PROJECT_VERSION_MAJOR}"
+             EXPORT_NAME example
+             OUTPUT_NAME example
 )
 
 target_include_directories(
-    example_example
-    ${example_warning_guard}
-    PUBLIC
-    "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
-    "$<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>"
+  example_example # XXX ${example_warning_guard}
+  PUBLIC "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
+         "$<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>"
+         "$<INSTALL_INTERFACE:${EXAMPLE_INCLUDE_DIRECTORY}>"
 )
 
-target_compile_features(example_example PUBLIC cxx_std_14)
+target_compile_features(example_example PUBLIC cxx_std_17)
 
 # ---- Install ----
 
 include(CMakePackageConfigHelpers)
-include(GNUInstallDirs)
-
-set(example_include_directory "${CMAKE_INSTALL_INCLUDEDIR}/example")
 
 install(
-    DIRECTORY
-    "${PROJECT_SOURCE_DIR}/include/"
-    "${PROJECT_BINARY_DIR}/include/"
-    DESTINATION "${example_include_directory}"
-    COMPONENT example_Development
+  DIRECTORY "${PROJECT_SOURCE_DIR}/include/" "${PROJECT_BINARY_DIR}/include/"
+  DESTINATION "${EXAMPLE_INCLUDE_DIRECTORY}"
+  COMPONENT example_Development
 )
 
 install(
-    TARGETS example_example
-    EXPORT exampleTargets
-    RUNTIME #
-    COMPONENT example_Runtime
-    LIBRARY #
-    COMPONENT example_Runtime
-    NAMELINK_COMPONENT example_Development
-    ARCHIVE #
-    COMPONENT example_Development
-    INCLUDES #
-    DESTINATION "${example_include_directory}"
+  TARGETS example_example
+  EXPORT exampleTargets
+  RUNTIME COMPONENT example_Runtime
+  LIBRARY COMPONENT example_Runtime NAMELINK_COMPONENT example_Development
+  ARCHIVE COMPONENT example_Development
+  INCLUDES
+  DESTINATION "${EXAMPLE_INCLUDE_DIRECTORY}"
 )
 
 write_basic_package_version_file(
-    exampleConfigVersion.cmake
-    COMPATIBILITY SameMajorVersion
+  exampleConfigVersion.cmake COMPATIBILITY SameMajorVersion
 )
 
-set(
-    example_INSTALL_CMAKEDIR "${CMAKE_INSTALL_LIBDIR}/cmake/example"
+set(EXAMPLE_INSTALL_CMAKEDIR
+    "${CMAKE_INSTALL_LIBDIR}/cmake/example"
     CACHE STRING "CMake package config location relative to the install prefix"
 )
-mark_as_advanced(example_INSTALL_CMAKEDIR)
+mark_as_advanced(EXAMPLE_INSTALL_CMAKEDIR)
```

Comment:
Why change the casing? The GUI CMake clients will still group variables properly regardless of casing, these are not variables that go into configured header files for macro names and the `project` call was given `example` as a name not `EXAMPLE`.